File: db_sqlite/lang.go
package main

import (
	"database/sql"
	_ "github.com/mattn/go-sqlite3"
    "math/rand"
    "os"
    "time"
    "fmt"
)

var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")

func RandSeq(n int) string {
    b := make([]rune, n)
    for i := range b {
        b[i] = letters[rand.Intn(len(letters))]
    }
    return string(b)
}

type Lang struct {
    name string
    desc string
    num int
    percent float64
}

func main() {

    os.Remove("./lang.db")

    db, _ := sql.Open("sqlite3", "./lang.db")
	defer db.Close()

	sqlStmt := "create table lang (id integer not null primary key, name text, desc text, num integer, percent real);"
	db.Exec(sqlStmt)

    data := make([]Lang, 1000000)
    for i := range data {
        data[i] = Lang{RandSeq(10), RandSeq(10), 10+rand.Intn(5000000), 100.0*rand.Float64()}
    }

    tx, _ := db.Begin()
    stmt, _ := tx.Prepare("insert into lang (name, desc, num, percent) values (?, ?, ?, ?)")
	defer stmt.Close()

    begin := time.Now()

    for _, l := range data {
		stmt.Exec(l.name, l.desc, l.num, l.percent)
	}

	tx.Commit()
    fmt.Println("Executing: ", time.Since(begin))
}

File: db_sqlite/lang.py
import sqlite3
import random
import string
import os
import time

LETTERS = string.lowercase + string.uppercase + string.digits
re = random.Random()

def randstr(n):
    return "".join(re.choice(LETTERS) for _ in xrange(n))

try:
    os.remove('./lang.db')
except OSError:
    pass

conn = sqlite3.connect('./lang.db')
c = conn.cursor()
c.execute("create table lang (name text, desc text, num integer, percent real)")
conn.commit()

#data = [(randstr(10), randstr(10), re.randint(10, 5000000), re.uniform(0, 100)) for _ in xrange(1000000)]

data = [("Java", "Bla", 123445, 34535.345) for _ in xrange(1000000)]
begin = time.clock()

c.executemany("insert into lang values (?, ?, ?, ?)", data)
conn.commit()
conn.close()

end = time.clock()
print "Executing:", (end - begin), "s"
